core/src/shred_fetch_stage.rs

```diff
@@ -14,9 +16,54 @@ pub struct ShredFetchStage {
 }
 
 impl ShredFetchStage {
+    // updates packets received on a channel and sends them on another channel
+    fn modify_packets<F>(recvr: PacketReceiver, sendr: PacketSender, modify: F)
+    where
+        F: Fn(&mut Packet),
+    {
+        while let Some(mut p) = recvr.iter().next() {
+            p.packets.iter_mut().for_each(|p| modify(p));
+            if sendr.send(p).is_err() {
+                break;
```

Comment:
Why not try to send the rest of the packets?

Reply:
If the sender fails there's no point, it won't work again. iirc it only fails if the channel is disconnected.